src/storage/backend/backend_impl.rs

```diff
@@ -178,9 +200,10 @@ impl Storage for Backend {
         self.operator
             .remove_all(&get_file_path(ino))
             .await
-            .unwrap_or_else(|e| {
-                panic!("Failed to remove a file from the backend of ino={ino}: {e}")
-            });
+            .map_err(|e| StorageError {
```

Comment:
Use `?` instead of `map_err`.

Reply:
`e` cannot be converted into `StorageError` directly, because we must fill the field `operation` manually.

Reply:
I feel like there's no need to store operation info in the error. If we need that info, we could just let the caller use `error!` to log it. The thing is, `map_err` is kind of spreading all over the code, and it doesn’t look too pretty.

Reply:
Ok, I will fix this later.

Reply:
Maybe we need a `BackEndError` ?

Reply:
> Maybe we need a `BackEndError` ?

What does it contain? I think `StorageErrorInner` (which will be renamed to `StorageError`) is enough, it contains an opendal error, with the context of the backend operation.

Reply:
Since memory cache is currently error free. The StorageErorr is equal to opendal::error ? If so ,just use the opendal::error. 
But when disk cache is supported, cache will gen error also，which might be `std::io::errror`.

Reply:
> Since memory cache is currently error free. The StorageErorr is equal to opendal::error ? If so ,just use the opendal::error. But when disk cache is supported, cache will gen error also，which might be `std::io::errror`.

opendal operations will also return `std::io::Error`, for example, in `Writer`s and `Reader`s. Therefore, the enum of:

- `std::io::Error`
- `opendal::Error`
- `OutOfRange` (for blocks)
- `Internal` (anyhow errors)

is enough and not redundent. I will use this enum directly, so that we can use `?` here.